unic/ucd/normal/src/decomposition_type.rs

```diff
@@ -12,228 +12,180 @@
+        Canonical {
+            abbr => Can,
+            long => Canonical,
+            human => "Canonical",
+        }
 
-/// Represents the Unicode character
-/// [*Decomposition_Type*](http://www.unicode.org/reports/tr44/#Decomposition_Type) property.
-///
-/// * <http://www.unicode.org/reports/tr44/#Character_Decomposition_Mappings>
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-#[allow(missing_docs)]
-pub enum DecompositionType {
-    Canonical,
-    Compat,
-    Circle,
-    Final,
-    Font,
-    Fraction,
-    Initial,
-    Isolated,
-    Medial,
-    Narrow,
-    Nobreak,
-    None,
-    Small,
-    Square,
-    Sub,
-    Super,
-    Vertical,
-    Wide,
-}
-
+        /// Otherwise unspecified compatibility character
+        Compat {
+            abbr => Com,
+            long => Compat,
+            human => "Unspecified",
+        }
 
-impl CharProperty for DecompositionType {
-    fn prop_abbr_name() -> &'static str {
-        "dt"
-    }
+        /// Encircled form
+        Circle {
+            abbr => Enc,
+            long => Circle,
+            human => "Encircled",
+        }
 
-    fn prop_long_name() -> &'static str {
-        "Decomposition_Type"
-    }
+        /// Final presentation form (Arabic)
+        Final {
+            abbr => Fin,
+            long => Final,
+            human => "Arabic Final",
+        }
 
-    fn prop_human_name() -> &'static str {
-        "Decomposition Type"
-    }
-}
+        /// Font variant (for example, a blackletter form)
+        Font {
+            abbr => Font,
+            long => Font,
+            human => "Font Variant",
+        }
 
+        /// Vulgar fraction form
+        Fraction {
+            abbr => Fra,
+            long => Fraction,
+            human => "Vulgar Fraction",
+        }
 
-impl PartialCharProperty for DecompositionType {
-    fn of(ch: char) -> Option<Self> {
-        Self::of(ch)
-    }
-}
+        /// Initial presentation form (Arabic)
+        Initial {
+            abbr => Init,
+            long => Initial,
+            human => "Arabic Initial",
+        }
 
+        /// Isolated presentation form (Arabic)
+        Isolated {
+            abbr => Iso,
+            long => Isolated,
+            human => "Arabic Isolated",
+        }
 
-impl EnumeratedCharProperty for DecompositionType {
-    fn all_values() -> &'static [Self] {
-        Self::all_values()
-    }
+        /// Medial presentation form (Arabic)
+        Medial {
+            abbr => Med,
+            long => Medial,
+            human => "Arabic Medial",
+        }
 
-    fn abbr_name(&self) -> &'static str {
-        self.abbr_name()
-    }
+        /// Narrow (or hankaku) compatibility character
+        Narrow {
+            abbr => Nar,
+            long => Narrow,
+            human => "Narrow",
+        }
 
-    fn long_name(&self) -> &'static str {
-        self.long_name()
-    }
+        /// No-break version of a space or hyphen
+        NoBreak {
```

Comment:
(I think you missed a few words, I can't tell if that's slightly towards or against.)

Here's my main logic:

- The [formatting tag](http://unicode.org/reports/tr44/#Formatting_Tags_Table) is `<noBreak>`
  - (This would allow dropping the `to_lowercase` call in `unic-gen::generate::ucd::normal::CompatibilityDecompositionData::emit`, though I haven't here and used the `long_names` module instead)
- It's described as "No-break" (two words)
- My spell checker gets annoyed about "Nobreak"
  - It fits Rust conventions of `CamelCase`; `Nobreak` feels like `Runoncase` because conceptually it is the "no-break" type and not the "nobreak" type

I thought I'd change this here and see what your opinion was here (good on you to catch it; I was actually going to mention it but I forgot), as now we have access to it as `Nobreak` in `mod long_names`.